public/app/features/dashboard-scene/settings/VersionsEditView.tsx

```diff
@@ -92,6 +96,27 @@ export class VersionsEditView extends SceneObjectBase<VersionsEditViewState> imp
     return sceneGraph.getTimeRange(this._dashboard);
   }
 
+  public async onRestore(version: DecoratedRevisionModel): Promise<boolean> {
+    const versionRsp = await historySrv.restoreDashboard(version.uid, version.version);
+
+    if (!isNumber(versionRsp.version)) {
+      return false;
+    }
+
+    const dashboardDTO: DashboardDTO = {
+      dashboard: new DashboardModel(version.data),
+      meta: this._dashboard.state.meta,
+    };
+    const dashScene = transformSaveModelToScene(dashboardDTO);
+    const newState = sceneUtils.cloneSceneObjectState(dashScene.state);
+    newState.version = versionRsp.version;
+
+    this._dashboard.setInitialState(newState);
+    this._dashboard.onDiscard();
+
+    return true;
```

Comment:
Think this logic could belong to `DashboardScene.onRestore`. Then you wouldn't have to expose setter for the initialState.